crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -315,6 +477,141 @@ impl<F, T>
                     ..item.data
                 })
             }
+            BraintreePaymentsResponse::ClientTokenResponse(client_token_data) => Ok(Self {
+                status: enums::AttemptStatus::AuthenticationPending,
+                response: Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::NoResponseId,
+                    redirection_data: Some(services::RedirectForm::Braintree {
+                        client_token: client_token_data
+                            .data
+                            .create_client_token
+                            .client_token
+                            .expose(),
+                        card_token: item.data.get_payment_method_token()?,
+                        bin: match item.data.request.payment_method_data.clone() {
+                            api_models::payments::PaymentMethodData::Card(card_details) => {
+                                card_details.card_number.get_card_isin()
+                            }
+                            api_models::payments::PaymentMethodData::CardRedirect(_)
+                            | api_models::payments::PaymentMethodData::Wallet(_)
+                            | api_models::payments::PaymentMethodData::PayLater(_)
+                            | api_models::payments::PaymentMethodData::BankRedirect(_)
+                            | api_models::payments::PaymentMethodData::BankDebit(_)
+                            | api_models::payments::PaymentMethodData::BankTransfer(_)
+                            | api_models::payments::PaymentMethodData::Crypto(_)
+                            | api_models::payments::PaymentMethodData::MandatePayment
+                            | api_models::payments::PaymentMethodData::Reward
+                            | api_models::payments::PaymentMethodData::Upi(_)
+                            | api_models::payments::PaymentMethodData::Voucher(_)
+                            | api_models::payments::PaymentMethodData::GiftCard(_) => {
+                                Err(errors::ConnectorError::NotImplemented(
+                                    "given payment method".to_owned(),
+                                ))?
+                            }
+                        },
+                        amount: item.data.request.amount,
+                    }),
+                    mandate_reference: None,
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                }),
+                ..item.data
+            }),
```

Comment:
Move this to try_from and use in BraintreePaymentsResponse and BraintreeAuthResponse try_from conversion